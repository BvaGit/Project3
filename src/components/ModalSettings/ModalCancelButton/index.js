import { connect } from 'react-redux';

import ModalCancelButton from "../ModalSettingsButton";
import { changeRegistrationField } from '../../../store/registration/actions';

const mapDispatchToProps = (dispatch) => ({
    changeField:(data) => dispatch(changeRegistrationField(data)),
})

export default connect(null, mapDispatchToProps)(ModalCancelButton);
