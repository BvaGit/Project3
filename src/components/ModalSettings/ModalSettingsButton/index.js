import { connect } from 'react-redux';

import ModalSettingsButton from './ModalSettingsButton';
import { ChangeCredits } from '../../../store/user/asyncActions';
import {getIsValidRegFields} from "../../../store/registration/selectors";

const mapStateToProps = (state) => ({
    isError: getIsValidRegFields(state),
});
const mapDispatchToProps = (dispatch) => ({
    changeLogin: () => dispatch(ChangeCredits())
})

export default connect(mapStateToProps, mapDispatchToProps)(ModalSettingsButton);
