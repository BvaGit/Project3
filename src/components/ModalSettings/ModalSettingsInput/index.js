import ModalSettingsInput from "./ModalSettingsInput";
import { connect } from 'react-redux';
import { getSettingsLoginStore, getSettingsPasswordStore } from '../../../store/settings/selectors';
import { handleValueInput } from '../../../store/settings/actions';

const mapStateToProps = (state, props) => ({
    login: getSettingsLoginStore(state, props),
    password: getSettingsPasswordStore(state, props)
})
const mapDispatchToProps = (dispatch) => ({
    changeInput: (data) => dispatch(handleValueInput(data))
})

export default connect(mapStateToProps, mapDispatchToProps)(ModalSettingsInput);
