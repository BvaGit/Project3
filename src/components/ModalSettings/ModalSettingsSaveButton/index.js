import { connect } from 'react-redux';

import ModalSettingsSaveButton from './ModalSettingsSaveButton';
import { ChangeCredits } from '../../../store/users/asyncActions';

const mapDispatchToProps = (dispatch) => ({
    changeLogin: () => dispatch(ChangeCredits())
})

export default connect(null, mapDispatchToProps)(ModalSettingsSaveButton);
