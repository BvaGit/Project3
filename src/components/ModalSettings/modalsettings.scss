@import '../../utils/variables.scss';

.content{
    @include flex(column,_,_);
    font-size: 18px;
    &__item-header{
        margin-right: $margin_md;
        margin-bottom: $margin_md;
    }
}
.content >div{
    margin: $margin_sm 0;
}
.content-language{
    display: flex;
    &__child{
        margin: auto $margin_md auto 3px
    }
}
.content-theme{
    @include flex(_,_,_);
    &__child{
        margin: auto $margin_md auto 3px
    }
}
.content-reset__header{
    margin-right: $margin_md;
}
.button-box{
    @include flex(_,space-between,_);
}
.button {
    width: 45%;
    height: 35px;
    margin: $margin_md 0;
    cursor: pointer;
    background: $button-color;
    color: $button-text;
    &:hover {
        background: $button-hover;
    }
}