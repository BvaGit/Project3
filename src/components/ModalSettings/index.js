import ModalSettings from "./ModalSettings";
import { connect } from 'react-redux';
import { ChangeCredits } from '../../store/users/asyncActions';

const mapStateToProps = (state) => ({});
const mapDispatchToProps = (dispatch) => ({
    changeLogin: (body) => dispatch(ChangeCredits(body))
});

export default connect(mapStateToProps, mapDispatchToProps)(ModalSettings);