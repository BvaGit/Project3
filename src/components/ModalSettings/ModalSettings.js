import React, { useState } from 'react';
import { FormattedMessage } from 'react-intl';

import Input from '../Input';

import "./modalsettings.scss";

function ModalSettings({languageValue: langValue, handleLanguage, closeModal, changeLogin}){
    const [login, setLogin] = useState('');
    const [password, setPassword] = useState('');


    const handleValueInput = (e) => {
        const body = {
            login, 
            password
        }
        changeLogin(body);
        closeModal();
    };
    const handleValueInputLogin = (e) => {
        setLogin(e.target.value)
    };
    const handleValueInputPassword = (e) => {
        setPassword(e.target.value)
    };

    const handleChange = (e) => {
        handleLanguage(e.target.value);
    };
    const handleThemeChange = (e) => {
        const selectedTheme = e.target.value;
        if (selectedTheme === 'light') {
            document.getElementById('root').classList.remove('theme-dark');
            document.getElementById('portal').classList.remove('theme-dark');
        } else {
            document.getElementById('root').classList.add('theme-dark');
            document.getElementById('portal').classList.add('theme-dark');
        }
    };

    return(
    
        <div className="content">
            <div  className="content-language">
                <div id="language" className="content__item-header">
                    <FormattedMessage 
                    id="language" 
                    defaultMessage="Language:" 
                    value={{ langValue }}>
                    </FormattedMessage>
                </div>
                    <Input labelId="en" id="en" name="language" type="radio" value="en" onChange={handleChange} className="content-language__child " />
                    <Input labelId="ru" id="ru" name="language" type="radio" value="ru" onChange={handleChange} className="content-language__child "/> 
            </div>
            <div className="content-theme">       
                <div id="theme" className="content__item-header">
                    <FormattedMessage 
                        id="theme" 
                        defaultMessage="Theme:" 
                        value={{ langValue }}>
                    </FormattedMessage>
                </div>
                    <Input id="light" labelId="light" name="theme" type="radio" value="light" onChange={handleThemeChange} className="content-theme__child" />
                    <Input id="dark" labelId="dark" name="theme" type="radio" value="dark" onChange={handleThemeChange} className="content-theme__child"/> 
            </div>
            <div className="content-reset">
                <span className="content__item-header content-reset__header">
                    <FormattedMessage 
                        id="notifications" 
                        defaultMessage="Notifications:" 
                        value={{ langValue }}>
                    </FormattedMessage>
                </span>
                <input type="checkbox"/>
            </div>
            <div>
                <button className="button">
                <FormattedMessage 
                    id="reset" 
                    defaultMessage="Reset all settings" 
                    value={{ langValue }}>
                </FormattedMessage>
                </button>
            </div>
            <div>                
                <Input id="newLogin" name="login" labelId="newLogin" type="text" placeholderId="newLoginPlaceholder" onChange={handleValueInputLogin}  />
            </div>
            <div>
                <Input id="newPassword" name="password" labelId="newPassword" type="password" placeholderId="newPasswordPlaceholder"
                onChange={handleValueInputPassword}  />
            </div>
            <div>
                <Input id="confirmPassword" name="confirmPassword" labelId="confirmPassword" type="password" placeholderId="repeatPasswordPlaceholder"  /> 
            </div>
            <div className="button-box">
                <button className="button" onClick={handleValueInput}>
                <FormattedMessage 
                    id="confirmBTN" 
                    defaultMessage="Confirm" 
                    value={{ langValue }}>
                </FormattedMessage>
                </button>
                <button className="button" onClick={closeModal}>
                <FormattedMessage 
                    id="cancelBTN" 
                    defaultMessage="Cancel" 
                    value={{ langValue }}>
                </FormattedMessage>
                </button>
            </div>
        </div>
    )
}

export default ModalSettings;
