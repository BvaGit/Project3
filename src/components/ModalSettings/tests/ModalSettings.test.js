import React from 'react';
import { shallow, mount } from 'enzyme';
import ModalSettings from '../ModalSettings';
import { IntlProvider } from "react-intl";
import { Provider } from 'react-redux';


describe('ModalSettings', () => {
    it("Should match snapshot", () => {
        const component = shallowBiggerSmart(<ModalSettings />)
        expect(component.html()).toMatchSnapshot()
    })
})

export const shallowBiggerSmart = (component, store) => {
    const core = store
        ? <Provider store={store}>{component}</Provider>
        : component;
    return shallow(<IntlProvider>core</IntlProvider>);
};