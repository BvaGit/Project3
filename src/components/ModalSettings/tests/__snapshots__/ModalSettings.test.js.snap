// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalSettings Should match snapshot 1`] = `"core"`;
